proposals/apple-mdm/base-setup-clix.md

```diff
@@ -0,0 +1,176 @@
+# MDM Base Setup
+
+Note: these can all be namespaced behind something like `apple-mdm` if needed. But I would like to avoid creating an mdm vertical vs. an osquery vertical to create an integrated product.
+
+## APNS
+### Step 1 - Generate CSR
+```
+$ fleetctl setup apns init
+Generating APNS keys at ./
+  fleet-apns-pki.crt ... done
+  fleet-apns-pki.key ... done
+  fleet-apns-push.key ... done
+  fleet-apns-push.csr ... done
+Send the .csr and .crt files to your friendly Fleet representative
+
+Keep the .key files. You will need them to complete this process. 
+
+Fleet will return a .p7m file to you within (TODO) business days. Use that file in the `fleetctl setup apns gen-req` command.
+```
+
+### Step 2 - Generate .req file from .p7m
+```
+$ fleetctl setup apns gen-req \
+--p7-file=~/path/to/*.p7 \
+--pki-key=~/path/to/fleet-apns-pki.key \
+--pki-crt=~/path/to/fleet-apns-pki.crt
+Working...
+Generated fleet-apns-push.req file at ./
+
+Visit identity.apple.com to upload this .req file. For more details, follow these instructions: NEED LINK HERE
+
+Apple will generate a .crt file. Use that file in the `fleetctl setup apns finalize` command. 
+```
+
+### Step 3 - Upload push..crt and push.key files to Fleet ENV
+
+```sh
+$ fleetctl setup apns finalize
+Update your Fleet server's environment variables as follows:
+1. Update ENV["APPLE_MDM_ENABLE"]=1
```

Comment:
@lucasmrod WDYT of these env var names? I simplified them a bit. Also the SCEP ones below.

Reply:
- `FLEET_` prefix is required for consistency with all the other flags.
- `APPLE_MDM_*` looks good to me (I just have to swap the keywords in the PoC).

PS: You can configure Fleet with env vars (1), cli flags (2) or a yaml (3). (It's Fleet's standard way of passing configuration to `fleet serve`). Meaning you can set your config the following three ways:

1. `FLEET_APPLE_MDM_ENABLE=1 [...] fleet serve`
2. `fleet serve [...] --apple-mdm-enable [...]`
3. `fleet serve --config config.yaml` # where the yaml contains:
```
---
apiVersion: v1
kind: config
spec:
  apple:
    mdm:
      enable: true
[...]
```

Reply:
> You can configure Fleet with env vars (1), cli flags (2) or a yaml (3). (It's Fleet's standard way of passing configuration to fleet serve)

@lucasmrod this is correct but I'm not sure if you use the `config` YAML document for these. 

It looks like the YAML config file used to pass configuration to `fleet serve` is a different YAML document:

![Screen Shot 2022-09-02 at 3 24 17 PM](https://user-images.githubusercontent.com/47070608/188223345-6d8e0424-f255-4043-9f7f-d9cc15935ef4.png)

Reply:
I will update the env vars to include FLEET_.

Do we have some doc somewhere that describes how to change the environment variables? I couldn't find it. If not, we can make it a requirement here to write one.

Reply:
> this is correct but I'm not sure you use the config YAML document for this.

Agree, was just mentioned as an "FYI".

We have builtin functionality to define configuration for Fleet, which automatically allows them to be configured in the mentioned ways, `FLEET_FOO` env variables, cli flags `--foo` and in the yaml.

Reply:
> Do we have some doc somewhere that describes how to change the environment variables? I couldn't find it. If not, we can make it a requirement here to write one.

Env vars cannot be changed, they can only be set at startup:
```sh
FLEET_FOO=X fleet serve
```
If you want to change `FLEET_FOO` to `Y`, you have to kill the server and restart with the new value:
```sh
FLEET_FOO=Y fleet serve
```

Not sure if we have docs for such a thing. @ksatter

Reply:
> Do we have some doc somewhere that describes how to change the environment variables?

I think you're looking for the configuration docs: https://fleetdm.com/docs/deploying/configuration#configuration

These docs mention the 3 methods (env vars, CLI flags, and YAML): https://fleetdm.com/docs/deploying/configuration#options